### DISCUSSION TITLE
Support for installation on windows

### DISCUSSION THREAD

**Discussion by WajahatRiaz** [zephyrproject-rtos/zephyr#51039] [Q&A] [answered]
**Describe the bug**
`-- Configuring incomplete, errors occurred!` while building a blinky sample for the nucleo board.  The link to the supported board is as follows:
[https://docs.zephyrproject.org/latest/boards/arm/nucleo_f103rb/doc/index.html](url)

**To Reproduce**
`cd %HOMEPATH%\zephyrproject\zephyr`
`west build -p always -b nucleo_f103rb samples\basic\blinky`

**Expected behavior**
A successful build of the blinky sample.

**Impact**
Unable to build and test a zephyr based application.

**Logs and console output**
<img width="728" alt="image" src="https://user-images.githubusercontent.com/61377755/194229690-1ba4aa05-1734-4506-a12f-e3a0bde0a207.png">

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain: Zephyr SDK
 - Version used: Updated

**Additional context**
I have installed all dependencies from the documentaion given in the link below:
[https://docs.zephyrproject.org/latest/develop/getting_started/index.html](url)

**Comment by erwango** [comment]
Can you have a try with another board? 
Likely you aren't able to compile any board

From what I can see you have a space in your path ("C:\Users\Wajahat Riaz"). This is generally causing issues. I'd suggest to have a new try with a path that has no space in it.

**Comment by WajahatRiaz** [comment]
> Can you have a try with another board? 
> Likely you aren't able to compile any board
> 
> From what I can see you have a space in your path ("C:\Users\Wajahat Riaz"). This is generally causing issues. I'd suggest to have a new try with a path that has no space in it.

I tried it with another board and QEMU as well. Getting the same error message. I believe there might be an issue with the path. I will try a different one without spaces.

**Comment by erwango** [comment]
Thanks for the update. I'm moving this ticket to a discussion

**Reply by WajahatRiaz** [reply]
> Thanks for the update. I'm moving this ticket to a discussion

Hi @erwango, hope you're doing well. Changing the path resolved the issue but I ended up into another one shown below:
<img width="668" alt="image" src="https://user-images.githubusercontent.com/61377755/194419421-01ecec3e-f2f8-47f2-a0b5-e6453eaf5990.png">

**Reply by erwango** [reply]
Ok, so you need to get back to installation guide. I get you missed some `pip install ...`

**Reply by WajahatRiaz** [reply]
> Ok, so you need to get back to installation guide. I get you missed some `pip install ...`

@erwango I checked again, requirements were already installed. Still generating the same error. I'm using the guide from the link below:
[https://docs.zephyrproject.org/latest/develop/getting_started/index.html](url)

**Reply by erwango** [reply]
Lilely you had an issue when installing.
Run `pip install -r pyelftools`

**Reply by WajahatRiaz** [reply]
> Lilely you had an issue when installing. Run `pip install -r pyelftools`

@erwango I tried it as well. Still the issue persists. As you can see below that it was installed.
<img width="667" alt="image" src="https://user-images.githubusercontent.com/61377755/194527060-7d7dbccb-c06d-4cff-976a-ef7b229125c6.png">

**Reply by erwango** [reply]
Ok, I'm not working on Windows myself, maybe you can ask support on Discord. 
But from what I can see this has nothing to do with Zephyr itself but rater your own python set up.
Maybe you can find answers as well in with google.

**Reply by WajahatRiaz** [reply]
The main reason behind the issue was that I didn't add the environment variables. Once you add them please restart your machine. I wasted a lot of time over this.

**Comment by WajahatRiaz** [comment, marked as answer]
The main reason behind the issue was that I didn't add the environment variables. Once you add them please restart your machine. Thank you @erwango for your time and dedication.
